cmd/playground/window_model.go

```diff
@@ -0,0 +1,207 @@
+func (m *model) handleDecrease() {
+	switch m.cursor {
+	case pointToTruncate:
+		m.state.DecrementTruncate()
+	case pointToYear, pointToMonth, PointToDay, pointToHour, pointToMinute:
+		m.state.DecrementTime(m.cursor.getStringValue())
+	}
+}
+
+func (m *model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	// switch currMsg := msg.(type) {
+	// case tea.KeyMsg:
+	currMsg := reflect.TypeOf(msg)
+	if currMsg.String() == "tea.KeyMsg" {
+		switch fmt.Sprintf("%s", msg) {
+		case "ctrl+c", "q":
+			return m, tea.Quit
+		case "up":
+			m.handleUp()
+		case "down":
+			m.handleDown()
+		case "left":
+			m.handleLeft()
+		case "right":
+			m.handleRight()
+		case "shift+up", "shift+right":
+			m.handleIncrease()
+		case "shift+down", "shift+left":
+			m.handleDecrease()
+		case "M", "h", "-", "1", "2", "3", "4", "5", "6", "7", "8", "9", "0", "backspace":
+			// to get updated the input fields must be focussed since we handle the Focus and Blur we can directly update them here
+			m.sizeInput, _ = m.sizeInput.Update(msg)
+			m.offsetInput, _ = m.offsetInput.Update(msg)
+			m.state.updateWindowparameters(m.sizeInput.Value(), m.offsetInput.Value())
+		}
+	}
+	return m, nil
+}
+
+func (m *model) genarateCursorIfPointed(position string, value string) string {
+	if m.cursor.getStringValue() == position {
+		var s strings.Builder
+		s.WriteString("[")
+		s.WriteString(value)
+		s.WriteString("]")
+		return s.String()
+	}
+	return " " + value
+}
+
+// we genare a string representing the sechduled time , which also adds a cursor if it is pointing to the any of the fields in sechdueld date
+func (m *model) genarateSechduledDateView() string {
+	var s strings.Builder
+	s.WriteString(m.genarateCursorIfPointed(pointToHour.getStringValue(), strconv.Itoa(m.state.scheduledTime.Hour())))
+	s.WriteString(":")
+	s.WriteString(m.genarateCursorIfPointed(pointToMinute.getStringValue(), strconv.Itoa(m.state.scheduledTime.Minute())))
+	s.WriteString(m.genarateCursorIfPointed(PointToDay.getStringValue(), strconv.Itoa(m.state.scheduledTime.Day())))
+	s.WriteString(m.genarateCursorIfPointed(pointToMonth.getStringValue(), m.state.scheduledTime.Month().String()))
+	s.WriteString(m.genarateCursorIfPointed(pointToYear.getStringValue(), strconv.Itoa(m.state.scheduledTime.Year())))
+	return s.String()
+}
+
+// this will update the values of Size and offset of the window versions present in state to the new values taken from the input
+
+// this will  be invoked for every update
+func (m *model) View() string {
+	var s strings.Builder
+	s.WriteString("Size\t\t")
+	s.WriteString(m.sizeInput.View())
+	s.WriteString("\n")
+	s.WriteString("Offset\t\t")
+	s.WriteString(m.offsetInput.View())
+	s.WriteString("\n")
+	s.WriteString("TruncateTo\t ")
+	s.WriteString(m.genarateCursorIfPointed(pointToTruncate.getStringValue(), m.state.windowV2.TruncateTo))
+	s.WriteString("\n")
+	s.WriteString("sechduled date   ")
+	s.WriteString(m.genarateSechduledDateView())
+	s.WriteString("\n\n\n             ")
```

Comment:
try to use `\t` for any whitespace to make it easier to decide the number of whitespaces to be used.